protected/controllers/SiteController.php

```diff
@@ -2,10 +2,131 @@
 
 class SiteController extends CController
 {
+
+    public function actionSearch($name = null,
+	    $style = null, 
+	    $hop = null, 
+	    $yeast = null, 
+	    $malt = null,
+	    $lastBeerId = null,
+	    $firstPage = null,
+	    $message = null
+    )
+    {
+	//Pagination
+	$message = $firstPage;
+
+	$beers = self::searchBeers($name, $style, $hop, $yeast, $malt, $lastBeerId);
+
+	$this->render('search', compact('beers', 
+					'name', 
+					'style', 
+					'hop', 
+					'yeast', 
+					'malt', 
+					'firstPage',
+					'message'));
+    }
+
+    public function actionAddBeer()
+    {
+	$message = "";
+
+	$name = isset($_POST['name']) ? $_POST['name'] : null;
+	$style = isset($_POST['style']) ? $_POST['style'] : null;
+
+	if($name && $style){
+		$result = self::addBeer($name, $style);
+		if($result){
+			$message = "Successfully added \"$name\"";
+		} else {
+			$message = "Error: Unable to add beer with name \"$name\"";
+		}
+	}
+
+	$styles = Style::model()->findAll();
+	$this->render('add', compact('styles', 'name', 'style', 'message')); 
+    }
+
     public function actionIndex()
     {
         $beers = Beer::model()->findAll();
 
         $this->render('index', compact('beers'));
     }
+
+    public static function addBeer($name, $style)
+    {
+	$result = false;
+
+	$beer = new Beer();
+	$beer->name = $name;
+	$beer->styleId = $style;
+	
+	//Temp fix until database can be updated to have auto_increment id
+	$beer->beerId = substr(md5(rand()), 0, 7);
```

Comment:
This is admittedly ugly bit of code that is a temporary work around until a fundamental and necessary change can be made to the database.
Because beerId, the primary key of the Beer table, is a random strong and not an auto_incrementing integer it has to be set explicitly by the code rather than handled by the database. 

Because the project is at its infancy migrating the database to have this auto_incrementing primary key is much less dangerous than if the project was more mature. In my mind this database change would be higher priority than a search feature, but the two piece of work should not be in the same merge request.